src/maker/config.rs

```diff
@@ -49,3 +55,178 @@ impl MakerConfig {
+                maker_config_section.get("min_contract_reaction_time"),
+                default_config.min_contract_reaction_time,
+            )
+            .unwrap_or(default_config.min_contract_reaction_time),
+            min_size: parse_field(
+                maker_config_section.get("min_size"),
+                default_config.min_size,
+            )
+            .unwrap_or(default_config.min_size),
+        })
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::fs::{self, File};
+    use std::io::Write;
+    use std::path::PathBuf;
+
+    fn create_temp_config(contents: &str, file_name: &str) -> PathBuf {
+        let file_path = PathBuf::from(file_name);
+        let mut file = File::create(&file_path).unwrap();
+        writeln!(file, "{}", contents).unwrap();
+        file_path
+    }
+
+    fn remove_temp_config(path: &PathBuf) {
+        fs::remove_file(path).unwrap();
+    }
+
+    #[test]
+    fn test_valid_config() {
+        let contents = r#"
+            [maker_config]
+            port = 6102
+            heart_beat_interval_secs = 3
+            rpc_ping_interval_secs = 60
+            watchtower_ping_interval_secs = 300
+            directory_servers_refresh_interval_secs = 43200
+            idle_connection_timeout = 300
+            absolute_fee_sats = 1000
+            amount_relative_fee_ppb = 10000000
+            time_relative_fee_ppb = 100000
+            required_confirms = 1
+            min_contract_reaction_time = 48
+            min_size = 10000
+        "#;
+        let config_path = create_temp_config(contents, "valid_maker_config.toml");
+        let config = MakerConfig::new(Some(&config_path)).unwrap();
+        remove_temp_config(&config_path);
+
+        let default_config = MakerConfig::default();
+        assert_eq!(config, default_config);
+    }
+
+    #[test]
+    fn test_missing_fields() {
+        let contents = r#"
+            [maker_config]
+            port = 6103
+        "#;
+        let config_path = create_temp_config(contents, "missing_fields_maker_config.toml");
+        let config = MakerConfig::new(Some(&config_path)).unwrap();
+        remove_temp_config(&config_path);
+
+        assert_eq!(config.port, 6103);
+        assert_eq!(
+            MakerConfig {
+                port: 6102,
+                ..config
+            },
+            MakerConfig::default()
+        );
+    }
+
+    #[test]
+    fn test_incorrect_data_type() {
+        let contents = r#"
+            [maker_config]
+            port = "not_a_number"
+        "#;
+        let config_path = create_temp_config(contents, "incorrect_type_maker_config.toml");
+        let config = MakerConfig::new(Some(&config_path)).unwrap();
+        remove_temp_config(&config_path);
+
+        assert_eq!(config, MakerConfig::default());
+    }
+
+    #[test]
+    fn test_missing_file() {
+        let config = MakerConfig::new(None).unwrap();
+        assert_eq!(config, MakerConfig::default());
+    }
+}
```

Comment:
This isn't testing the situation. When you put `None`, it uses the default file path, which actually exists. Thus the test works. With the current config logic, if the file actually didn't exist, this would fail in error.

Reply:
My bad ...

Reply:
To simulate this, you would need to give a specific file path where the config doesn't exists.. We should also make the case where the default path also doesn't exist. It's possible but can be handled later. I think it's better to keep the default config files in the repo as a reference. So , trying to do that now would be more complicated.